src/sentry/south_migrations/0456_backfill_groupenvironment_first_seen.py

```diff
@@ -0,0 +1,1278 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+class Migration(DataMigration):
+
+    # Flag to indicate if this migration is too risky
+    # to run online and needs to be coordinated for offline
+    is_dangerous = True
+
+    def forwards(self, orm):
+        db.commit_transaction()
+        try:
+            self._forwards(orm)
+        except Exception:
+            db.start_transaction()
+            raise
+        db.start_transaction()
+
+    def _forwards(self, orm):
+        "Write your forwards methods here."
+        GroupTagValue = orm.GroupTagValue
+        GroupEnvironment = orm.GroupEnvironment
+        Environment = orm.Environment
+
+        qs = GroupEnvironment.objects.filter(first_seen__isnull=True)
+        for ge in RangeQuerySetWrapperWithProgressBar(qs):
+            try:
+                env_name = Environment.objects.filter(
```

Comment:
There are only 280k environments total. I think we could probably keep them all in memory as a dict of `{<id>: <name>}` and then we skip a query on each `GroupEnvironment `